File: python/fibo_itera.py
import timeit

# Iterative Fibonacci
def fibonacci_iterative(n):
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a

print(timeit.timeit("fibonacci_iterative(12)", globals=globals(), number=1000))

File: cpp/fibo_itera.cpp
#include <iostream>
using namespace std;

int fibonacci_iterative(int);

int main() {
    int n = 12;
    cout << "Fibonacci(" << n << ") = " << fibonacci_iterative(n) << endl;
    return 0;
}

int fibonacci_iterative(int n) {
    int a = 0, b = 1, c;
    for (int i = 1; i < n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}